src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -252,10 +269,8 @@ private int GetShardNumberForReplicationItem(TransactionOperationContext context
                 case DocumentReplicationItem:
                 case TimeSeriesDeletedRangeItem:
                 case TimeSeriesReplicationItem:
-                    return _parent.Context.GetShardNumberFor(context, _documentInfoHelper.GetDocumentId(item));
                 case RevisionTombstoneReplicationItem:
-                    DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Shiran, DevelopmentHelper.Severity.Normal, "Handle this (document Id does not exist)");
-                    throw new NotSupportedInShardingException("TODO: implement for sharding"); // revision tombstones doesn't contain any info about the doc. The id here is the change-vector of the deleted revision
+                    return _parent.Context.GetShardNumberFor(context, _documentInfoHelper.GetDocumentId(item));
```

Comment:
let's throw if `_documentInfoHelper.GetDocumentId(item)` is null

Reply:
done